### DISCUSSION TITLE
Multi-Select that defers onChange until after all desired items are selected.

### DISCUSSION THREAD

**Discussion by michaelsawyers** [ariakit/ariakit#3048] [Questions] [answered]
How might I create a MultiSelect that defers setting the selected items until the popover closes? I'm aware of the `setValueOnChange` but still uncertain how to go about this.

The issue on my end is that the api is called (with slow performance) on each selection.. so as a solution our UX folks want to only call the the api after user is done selecting all the desired items. 

Open to alternative suggestions or criticism of this approach. Thanks!

**Comment by diegohaz** [comment, marked as answer]
I believe it hinges on when you determine the user has finished selecting. Generally, you can wait until the selection popover is closed by using the [`onClose`](https://ariakit.org/reference/select-popover#onclose) prop.
